### PR TITLE
feat: save vitest state in global variable

### PR DESCRIPTION
**Pull request by poyoho**
fix: #663

save vitest state in global variable for the nice error message.

### CODE REVIEW

File: packages/vitest/globals.d.ts [hunk 1]
```diff
@@ -11,5 +11,6 @@ declare global {
   const afterAll: typeof import('vitest')['afterAll']
   const beforeEach: typeof import('vitest')['beforeEach']
   const afterEach: typeof import('vitest')['afterEach']
+  let __vitest_worker__: import('vitest').WorkerGlobalState
```
Comment (antfu) [inline packages/vitest/globals.d.ts:14]: We shouldn't expose this to user type. Let's have it in the source code which will be removed in build

**Comment by poyoho** [comment]
may be can break the change, only cache the process in the global variable and reset it in every test file exec

**Comment by Demivan** [comment]
I don't think [1538626](https://github.com/vitest-dev/vitest/pull/739/commits/1538626d2c7c0a6f58468670a1f0c26f5ca52f6e) is needed. Users can still break it by re-assigning properties on `process`. So better error message is enough

**Comment by poyoho** [comment]
ok!
but I don't know how re-assigning properties on process can break it. after run the test file always reset the process, and the test file are no preemption of `process` 🤔

**Pull request merged**